src/main/kotlin/apply/ui/admin/mail/MailsFormView.kt

```diff
@@ -81,4 +78,19 @@ class MailsFormView(
             UI.getCurrent().navigate(MailsView::class.java)
         }
     }
+
+    private fun createPreviewButton(): Button {
+        return createContrastButton("미리보기") {
+            val result = getDataFromMailForm()
+            val body = mailService.generateMailBody(result)
+            MailPreviewDialog(body)
+        }
+    }
+
+    private fun getDataFromMailForm(): MailData {
+        return mailForm.bindOrNull() ?: run {
+            createNotification("받는사람을 한 명 이상 지정해야 합니다.")
+            throw IllegalArgumentException()
```

Comment:
```suggestion
            throw IllegalStateException("받는사람을 한 명 이상 지정해야 합니다.")
```

c: IllegalArgumentException 보다는 IllegalStateException 같은게 더 적절하지 않을까 싶습니다!
추가로 예외처리 메시지를 추가하면 문제가 있을 때 로그로 원인을 파악하기 좋겠네요.

Reply:
옷,,, 이 예외는 받는 사람 값 입력을 안한거라 IllegalArgumentException이 어울린다고 생각했는데 State 가 더 적합한 이유가 무엇일까요?
예외 메시지는 추가 좋네요 ㅎㅎ